src/main/java/uk/gov/indexer/dao/CurrentKeysUpdateDao.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.indexer.dao;
+
+import org.skife.jdbi.v2.sqlobject.Bind;
+import org.skife.jdbi.v2.sqlobject.SqlQuery;
+import org.skife.jdbi.v2.sqlobject.SqlUpdate;
+
+interface CurrentKeysUpdateDao {
+    String CURRENT_KEYS_TABLE = "CURRENT_KEYS";
+
+
+    @SqlUpdate("CREATE TABLE IF NOT EXISTS " + CURRENT_KEYS_TABLE + " (ID INTEGER PRIMARY KEY, KEY VARCHAR UNIQUE)")
+    void ensureCurrentKeysTableExists();
+
+
+    @SqlUpdate("INSERT INTO " + CURRENT_KEYS_TABLE + "(ID, KEY) VALUES(:id, :key)")
+    int writeCurrentKey(@Bind("id") int id, @Bind("key") String key);
+
+    @SqlUpdate("UPDATE " + CURRENT_KEYS_TABLE + " SET ID=:id WHERE KEY=:key")
+    int updateCurrentKey(@Bind("id") int id, @Bind("key") String key);
+
+    @SqlQuery("SELECT CASE " +
+            "WHEN EXISTS(" +
+            "SELECT 1 FROM " + CURRENT_KEYS_TABLE + " WHERE KEY=:key" +
+            ") " +
+            "THEN(" +
+            "SELECT ID FROM " + CURRENT_KEYS_TABLE + " WHERE KEY=:key" +
+            ") " +
+            "ELSE -1 " +
+            "END")
+    int getCurrentIdForKey(@Bind("key") String key);
```

Comment:
I will suggest to do it in next PR if feasible. I don't like idea of adding dropwizard jdbi-java8 dependancy for this purpose. there is other solution which is not very simple and quick. Will look into it separately.